zhaquirks/xiaomi/aqara/ctrl_neutral.py

```diff
@@ -53,6 +62,73 @@
 class CtrlNeutral(XiaomiCustomDevice):
     """Aqara single and double key switch device."""
 
+    class BasicClusterDecoupled(BasicCluster):
+        """Adds attributes for decoupled mode."""
+
+        # Known Options for 'decoupled_mode_<button>':
+        # * 254 (decoupled)
+        # * 18 (relay controlled)
+        manufacturer_attributes = {
+            0xFF22: ("decoupled_mode_left", t.uint8_t),
+            0xFF23: ("decoupled_mode_right", t.uint8_t),
+        }
+
+    class WallSwitchOnOffCluster(OnOff, EventableCluster):
+        """WallSwitchOnOffCluster: fire events corresponding to press type."""
+
+        press_type = {
+            0x00: SHORT_PRESS,
+            0x01: SHORT_RELEASE,
+            0x02: DOUBLE_PRESS,
+        }
+        name = "WallSwitch_cluster"
+        ep_attribute = "WallSwitch_cluster"
+
+        def __init__(self, *args, **kwargs):
+            """Init."""
+            self.last_tsn = -1
+            super().__init__(*args, **kwargs)
+
+        manufacturer_server_commands = {
+            0xFD: ("press_type", (t.uint8_t,), False),
+        }
+
+        def handle_cluster_request(self, tsn, command_id, args):
+            """Handle press_types command."""
+            if tsn == self.last_tsn:
+                _LOGGER.debug("WallSwitch: ignoring duplicate frame")
+                return
+
+            self.last_tsn = tsn
+            _LOGGER.error("handle_cluster_request: %s, %s", command_id, args)
+            super().handle_cluster_request(tsn, command_id, args)
+            if command_id == 0xFD:
+                press_type = args[0]
+                self.listener_event(
+                    ZHA_SEND_EVENT, self.press_type.get(press_type, "unknown"), []
+                )
+
+    class CustomOnOffCluster(OnOffCluster):
+        """Fire ZHA events for on off cluster."""
+
+        cluster_id = OnOff.cluster_id
+
+        def __init__(self, *args, **kwargs):
+            """Init."""
+            self._current_state = {}
+            super().__init__(*args, **kwargs)
+
+        def _update_attribute(self, attrid, value):
+            _LOGGER.info("%s: %s", attrid, value)
+            if attrid == 0:
+                self._current_state = PRESS_TYPES.get(value)
+                event_args = {
+                    PRESS_TYPE: self._current_state,
+                    VALUE: value,
+                }
+                self.listener_event(ZHA_SEND_EVENT, COMMAND_CLICK, event_args)
+            super()._update_attribute(attrid, self._current_state)
```

Comment:
so, for any attribute update, you'll be sending the last received "press type", regardless of the what attribute is being updated/reported? This doesn't look correct to me, unless i'm missing something